action-server/tests/test_action_send_checkin.py

```diff
@@ -0,0 +1,101 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+from hashids import Hashids
+from rasa_sdk import Tracker
+from rasa_sdk.executor import CollectingDispatcher
+
+from actions.action_send_checkin import ActionSendCheckin
+from actions.lib.exceptions import (
+    InvalidExternalEventException,
+    ReminderNotFoundException,
+)
+
+CHECKIN_BASE_URL = "http://test.com/ci/"
+HASHIDS_SALT = "abcd1234"
+HASHIDS_MIN_LENGTH = 4
+hashids = Hashids(HASHIDS_SALT, min_length=HASHIDS_MIN_LENGTH)
+
+DEFAULT_PHONE_NUMBER = "91112223333"
+DEFAULT_FIRST_NAME = "George"
+DEFAULT_REMINDER_ID = 1
+DEFAULT_REMINDER_ID_HASH = hashids.encode(1)
+
+ENV = {
+    "REMINDER_ID_HASHIDS_SALT": HASHIDS_SALT,
+    "REMINDER_ID_HASHIDS_MIN_LENGTH": str(HASHIDS_MIN_LENGTH),
+    "DAILY_CHECKIN_BASE_URL": CHECKIN_BASE_URL,
+}
+
+
+def _create_tracker(
+    sender_id=DEFAULT_PHONE_NUMBER, reminder_id=DEFAULT_REMINDER_ID_HASH
+) -> Tracker:
+    return Tracker(
+        sender_id,
+        {},
+        {"entities": [{"entity": "metadata", "value": {"reminder_id": reminder_id}}]},
+        [],
+        False,
+        None,
+        {},
+        "action_listen",
+    )
+
+
+def _mock_reminder(mock_session_factory, phone_number):
+    reminder_mock = (
+        mock_session_factory.return_value.query.return_value.get.return_value
+    )
+    reminder_mock.get_first_name.return_value = DEFAULT_FIRST_NAME
+    reminder_mock.get_phone_number.return_value = phone_number
+
+
+class TestDailyCiEnrollForm(TestCase):
```

Comment:
This looks like badly done copy-paste to me...

Reply:
you got me...